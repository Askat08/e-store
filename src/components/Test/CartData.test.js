import React from "react";
import { shallow } from "enzyme";
import CartData from "../Cart/CartData";

describe("CartData component test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CartData />);
  });

  it("shallow CartData component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("div props should have className", () => {
    expect(
      wrapper
        .find("div")
        .at(0)
        .hasClass("container-fluid text-center d-none d-lg-block")
    ).toBeTruthy();
    expect(wrapper.find("div").at(1).hasClass("row")).toBeTruthy();
    expect(
      wrapper.find("div").at(2).hasClass("col-10 mx-auto col-lg-2")
    ).toBeTruthy();
    expect(
      wrapper.find("div").at(3).hasClass("col-10 mx-auto col-lg-2")
    ).toBeTruthy();
    expect(
      wrapper.find("div").at(4).hasClass("col-10 mx-auto col-lg-2")
    ).toBeTruthy();
    expect(
      wrapper.find("div").at(5).hasClass("col-10 mx-auto col-lg-2")
    ).toBeTruthy();
    expect(
      wrapper.find("div").at(6).hasClass("col-10 mx-auto col-lg-2")
    ).toBeTruthy();
    expect(
      wrapper.find("div").at(7).hasClass("col-10 mx-auto col-lg-2")
    ).toBeTruthy();
  });

  it("paragraph correct text and className", () => {
    expect(wrapper.find("p").at(0).hasClass("text-uppercase")).toBeTruthy();
    expect(wrapper.find("p").at(0).text()).toEqual("comics");
    expect(wrapper.find("p").at(1).hasClass("text-uppercase")).toBeTruthy();
    expect(wrapper.find("p").at(1).text()).toEqual("name");
    expect(wrapper.find("p").at(2).hasClass("text-uppercase")).toBeTruthy();
    expect(wrapper.find("p").at(2).text()).toEqual("price");
    expect(wrapper.find("p").at(3).hasClass("text-uppercase")).toBeTruthy();
    expect(wrapper.find("p").at(3).text()).toEqual("quantity");
    expect(wrapper.find("p").at(4).hasClass("text-uppercase")).toBeTruthy();
    expect(wrapper.find("p").at(4).text()).toEqual("remove");
    expect(wrapper.find("p").at(5).hasClass("text-uppercase")).toBeTruthy();
    expect(wrapper.find("p").at(5).text()).toEqual("total");
  });
});
