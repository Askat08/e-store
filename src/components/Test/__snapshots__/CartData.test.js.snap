// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartData component test shallow CartData component 1`] = `
<div
  className="container-fluid text-center d-none d-lg-block"
>
  <div
    className="row"
  >
    <div
      className="col-10 mx-auto col-lg-2"
    >
      <p
        className="text-uppercase"
      >
        comics
      </p>
    </div>
    <div
      className="col-10 mx-auto col-lg-2"
    >
      <p
        className="text-uppercase"
      >
        name
      </p>
    </div>
    <div
      className="col-10 mx-auto col-lg-2"
    >
      <p
        className="text-uppercase"
      >
        price
      </p>
    </div>
    <div
      className="col-10 mx-auto col-lg-2"
    >
      <p
        className="text-uppercase"
      >
        quantity
      </p>
    </div>
    <div
      className="col-10 mx-auto col-lg-2"
    >
      <p
        className="text-uppercase"
      >
        remove
      </p>
    </div>
    <div
      className="col-10 mx-auto col-lg-2"
    >
      <p
        className="text-uppercase"
      >
        total
      </p>
    </div>
  </div>
</div>
`;
